src/common/catalog/catalog-entity-registry.ts

```diff
@@ -1,10 +1,15 @@
-import { action, computed, observable, IObservableArray } from "mobx";
+import { action, computed, observable, IComputedValue, IObservableArray } from "mobx";
 import { CatalogEntity } from "./catalog-entity";
+import { iter } from "../utils";
 
 export class CatalogEntityRegistry {
-  protected sources = observable.map<string, IObservableArray<CatalogEntity>>([], { deep: true });
+  protected sources = observable.map<string, IComputedValue<CatalogEntity[]>>([], { deep: true });
 
-  @action addSource(id: string, source: IObservableArray<CatalogEntity>) {
+  @action addObservableSource(id: string, source: IObservableArray<CatalogEntity>) {
+    this.sources.set(id, computed(() => source));
+  }
```

Comment:
Not really relevant to this PR but I bring this up here because you are changing this code. I find the term source too vague. It's an "independent" or sourced list of entities, right? This makes it a little clearer, and we could do the same for the exposed API?
```suggestion
  @action addObservableSource(id: string, entities: IObservableArray<CatalogEntity>) {
    this.sources.set(id, computed(() => entities));
  }
```
or call it `addObservableEntitySource`?

Reply:
So these are used `catalogEntityRegistry.addObservableSource(...)`. Do we need to have the word "Entity" twice, namely: `catalogEntityRegistry.addObservableEntitySource(...)` ?

Reply:
> Do we need to have the word "Entity" twice

No 🙂